### PR TITLE
documenting adding a package locally CONFIG= ./.scripts/run_docker_build.sh call

### PR DESCRIPTION
**Pull request by ngam**
Essentially run_docker_build will default to linux-anvil-comp7 if shyaml isn't installed. Noting that in the docs.

```
./.scripts/run_docker_build.sh: line 24: shyaml: command not found
WARNING: DOCKER_IMAGE variable not set and shyaml not installed. Falling back to quay.io/condaforge/linux-anvil-comp7
latest: Pulling from condaforge/linux-anvil-comp7

```

PR Checklist:

- [ ] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [ ] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [ ] put any other relevant information below

### CODE REVIEW

**Comment by ngam** [comment]
Ooops. Will need to fix where the docs were edited.

**Review by isuruf — CHANGES_REQUESTED** [review]
It's not a must. We have new code that doesn't require shyaml

**Comment by ngam** [comment]
> It's not a must. We have new code that doesn't require shyaml

So, should we ignore this for now or should I just reword it to "should"? When is the new code coming btw?

**Comment by ngam** [comment]
Well, I guess adding shyaml doesn't actually help much anyway...

How can this work on Macs? `CONFIG=osx64 ./.scripts/run_docker_build.sh` just doesn't do the trick...

**Comment by isuruf** [comment]
> When is the new code coming btw?

It's already there. You probably haven't rerendered

**Comment by jakirkham** [comment]
> How can this work on Macs?

Can you please share more about what you are trying to do? Using Docker to build packages on macOS probably doesn't make sense (unless Linux packages are being built there). For macOS packages builds, one can run `.scripts/run_osx_build.sh` [as seen here]( https://github.com/conda-forge/julia-feedstock/blob/c88f6ac54543b91c01549b993d9ad2935b38a39a/.azure-pipelines/azure-pipelines-osx.yml#L29 ).

**Comment by ngam** [comment]
Sorry, I should've been clearer. I am trying to do this in staged-recipes.

**Comment by ngam** [comment]
So, I will just run this, right? https://github.com/conda-forge/staged-recipes/blob/main/.scripts/run_osx_build.sh

**Comment by ngam** [comment]
Also, the motivation: I have run into some minor issues when building locally (staged-recipes) vs building on the CI, so I am trying to make sure I have identical setups.

**Comment by ngam** [comment]
Well, either way... this section of the docs needs further refinement. For example, `./.scripts/run_osx_build.sh` doesn't really work and it asks for password etc (to mess with homebrew installation or something)

**Comment by ngam** [comment]
> > When is the new code coming btw?
> 
> It's already there. You probably haven't rerendered

How can we check if it is already present for staged-recipes? Because I am not finding that to be the case (could still be something on my side)

**Comment by ngam** [comment]
Not to keep commenting (sorry, I am figuring this out as I go), but the docs say

> You need to define an environment variable named ``CONFIG``. Its value must be
 the name of one of the three YAML configuration files in the ``.ci_support``
 directory (either ``linux64``, ``osx64``, or ``win64``). As an example, you can
 invoke the command as follows.

Which is simply not accurate, or am I still missing something?

**Comment by jakirkham** [comment]
It is possible the same effort has not been made with the staged-recipes script as in the feedstocks. Feel free to submit any tweaks in a PR

**Comment by ngam** [comment]
Also, we should enable running the cuda images locally as well

**Comment by ngam** [comment]
> It is possible the same effort has not been made with the staged-recipes script as in the feedstocks. Feel free to submit any tweaks in a PR

I am not really too concerned with the legit code stuff here --- I am merely talking about the documentation part. I will prepare a more thorough PR to spell out all these details; as part of that, I will recommend some code changes (though someone else will likely have to write them up)

**Comment by jakirkham** [comment]
Both of those things would be welcome. Thanks for poking around here 🙂

For suggested code changes would recommend a new issue.

As far as documentation goes it is worthwhile to document what we have now (even if it may be funky from a user perspective and maybe even because it is) so that we can help more people through the process.

**Comment by ngam** [comment]
> As far as documentation goes it is worthwhile to document what we have now (even if it may be funky from a user perspective and maybe even because it is) so that we can help more people through the process.

Yes, that's exactly my intention/plan 👍 Once I get this mxnet business out of the way, I will get to this. Please have a look when you have a chance --- I am running into some basic errors that I think someone with more knowledge will be able to figure out pretty quickly
